gui/connected_device.h

```diff
@@ -0,0 +1,39 @@
+#ifndef CONNECTED_DEVICE_H
+#define CONNECTED_DEVICE_H
+
+#include "../common/generated_libs/network_protocol/network_protocol.pb.h"
+#include <functional>
+
+// Represents a connection to the device running the controller.
+class ConnectedDevice {
```

Comment:
I made an interface because I expected that this would be implemented by an actual class (with initialization and state), in case we're talking to the device. Based on your other comment noting that this API should be replaced with basically bidi streaming, I think I should keep this a class, but change its API.

Reply:
OK, let's revisit pending the refactoring discussed below.